--- ./src/location/plugins/CMakeLists.txt	2024-10-11 11:04:34.000000000 +0300
+++ ../maplibre-native-qt_v3.0.0_Source_work/src/location/plugins/CMakeLists.txt	2025-12-12 15:03:00.414218261 +0200
@@ -68,6 +68,11 @@
     set_target_properties(${MLN_QT_GEOSERVICES_PLUGIN} PROPERTIES CXX_CLANG_TIDY "${CLANG_TIDY_COMMAND}")
 endif()
 
+file(RELATIVE_PATH OE_QMAKE_PLUGINS_INSTALL_PREFIX
+    "${CMAKE_INSTALL_PREFIX}"
+    "${OE_QMAKE_PATH_PLUGINS}"
+)
+
 # QtLocation plugin installation
 install(
     EXPORT ${MLN_QT_NAME}LocationPluginGeoServicesTargets
@@ -77,10 +82,10 @@
 install(
     TARGETS ${MLN_QT_GEOSERVICES_PLUGIN} ${GeoServicesPluginOutputTargets}
     EXPORT ${MLN_QT_NAME}LocationPluginGeoServicesTargets
-    ARCHIVE DESTINATION "plugins/geoservices"
-    LIBRARY DESTINATION "plugins/geoservices"
-    OBJECTS DESTINATION "plugins/geoservices"
-    RUNTIME DESTINATION "plugins/geoservices"
+    ARCHIVE DESTINATION "${OE_QMAKE_PLUGINS_INSTALL_PREFIX}/geoservices"
+    LIBRARY DESTINATION "${OE_QMAKE_PLUGINS_INSTALL_PREFIX}/geoservices"
+    OBJECTS DESTINATION "${OE_QMAKE_PLUGINS_INSTALL_PREFIX}/geoservices"
+    RUNTIME DESTINATION "${OE_QMAKE_PLUGINS_INSTALL_PREFIX}/geoservices"
 )
 
 # QtLocation QML extension plugin
@@ -205,6 +210,11 @@
     set_target_properties(${MLN_QT_QML_PLUGIN} PROPERTIES CXX_CLANG_TIDY "${CLANG_TIDY_COMMAND}")
 endif()
 
+file(RELATIVE_PATH OE_QMAKE_QML_INSTALL_PREFIX
+    "${CMAKE_INSTALL_PREFIX}"
+    "${OE_QMAKE_PATH_QML}"
+)
+
 # QtLocation QML extension plugin installation
 install(
     EXPORT ${MLN_QT_NAME}LocationPluginQmlTargets
@@ -214,21 +224,21 @@
 install(
     TARGETS ${MLN_QT_QML_PLUGIN} ${QmlPluginOutputTargets}
     EXPORT ${MLN_QT_NAME}LocationPluginQmlTargets
-    ARCHIVE DESTINATION "qml/MapLibre"
-    LIBRARY DESTINATION "qml/MapLibre"
-    OBJECTS DESTINATION "qml/MapLibre"
-    RUNTIME DESTINATION "qml/MapLibre"
+    ARCHIVE DESTINATION "${OE_QMAKE_QML_INSTALL_PREFIX}/MapLibre"
+    LIBRARY DESTINATION "${OE_QMAKE_QML_INSTALL_PREFIX}/MapLibre"
+    OBJECTS DESTINATION "${OE_QMAKE_QML_INSTALL_PREFIX}/MapLibre"
+    RUNTIME DESTINATION "${OE_QMAKE_QML_INSTALL_PREFIX}/MapLibre"
 )
 install(
     FILES
         "${CMAKE_CURRENT_BINARY_DIR}/MapLibre/qmldir"
-    DESTINATION "qml/MapLibre"
+    DESTINATION "${OE_QMAKE_QML_INSTALL_PREFIX}/MapLibre"
 )
 
 if(COMMAND qt_add_qml_module)
     install(
         FILES
             "${CMAKE_CURRENT_BINARY_DIR}/MapLibre/${MLN_QT_QML_PLUGIN}.qmltypes"
-        DESTINATION "qml/MapLibre"
+        DESTINATION "${OE_QMAKE_QML_INSTALL_PREFIX}/MapLibre"
     )
 endif()
