diff --git a/libosmscout-client-qt/src/osmscoutclientqt/DBJob.cpp b/libosmscout-client-qt/src/osmscoutclientqt/DBJob.cpp
index 55f2b7335..61a73b61a 100644
--- a/libosmscout-client-qt/src/osmscoutclientqt/DBJob.cpp
+++ b/libosmscout-client-qt/src/osmscoutclientqt/DBJob.cpp
@@ -107,7 +107,7 @@ void DBLoadJob::Run(const osmscout::BasemapDatabaseRef& basemapDatabase,
     }
     osmscout::GeoBox dbBox=db->GetDBGeoBox();
     if (!dbBox.Intersects(lookupBox)){
-      qDebug() << "Skip database" << db->path;
+      //qDebug() << "Skip database" << db->path;
       continue;
     }
     relevantDatabases.push_back(db);
diff --git a/libosmscout-client-qt/src/osmscoutclientqt/DBThread.cpp b/libosmscout-client-qt/src/osmscoutclientqt/DBThread.cpp
index ba993c59c..ae59610b7 100644
--- a/libosmscout-client-qt/src/osmscoutclientqt/DBThread.cpp
+++ b/libosmscout-client-qt/src/osmscoutclientqt/DBThread.cpp
@@ -271,7 +271,7 @@ void DBThread::onDatabaseListChanged(QList<QDir> databaseDirectories)
 
     if (database->Open(basemapLookupDirectory.toLocal8Bit().data())) {
       basemapDatabase=database;
-      qDebug() << "Basemap found and loaded from '" << basemapLookupDirectory << "'...";
+      //qDebug() << "Basemap found and loaded from '" << basemapLookupDirectory << "'...";
     }
     else {
       qWarning() << "Cannot open basemap database '" << basemapLookupDirectory << "'!";
diff --git a/libosmscout-client-qt/src/osmscoutclientqt/MapManager.cpp b/libosmscout-client-qt/src/osmscoutclientqt/MapManager.cpp
index 0aa85c4ac..2fb8ea3f3 100644
--- a/libosmscout-client-qt/src/osmscoutclientqt/MapManager.cpp
+++ b/libosmscout-client-qt/src/osmscoutclientqt/MapManager.cpp
@@ -236,7 +236,7 @@ bool MapDirectory::deleteDatabase()
 MapManager::MapManager(QStringList databaseLookupDirs, SettingsRef settings):
   databaseLookupDirs(databaseLookupDirs)
 {
-  qDebug() << "MapManager ctor";
+  //qDebug() << "MapManager ctor";
   webCtrl.setCookieJar(new PersistentCookieJar(settings));
   // we don't use disk cache here
 }
diff --git a/libosmscout-client-qt/src/osmscoutclientqt/MapRenderer.cpp b/libosmscout-client-qt/src/osmscoutclientqt/MapRenderer.cpp
index 90eb3bca5..fdfc4f683 100644
--- a/libosmscout-client-qt/src/osmscoutclientqt/MapRenderer.cpp
+++ b/libosmscout-client-qt/src/osmscoutclientqt/MapRenderer.cpp
@@ -317,7 +317,7 @@ void DBRenderJob::Run(const osmscout::BasemapDatabaseRef& basemapDatabase,
     }
 
     if (skip) {
-      osmscout::log.Debug() << "Skip database " << db->path.toStdString();
+      //osmscout::log.Debug() << "Skip database " << db->path.toStdString();
       continue;
     }
 
diff --git a/libosmscout-map/src/osmscoutmap/oss/Parser.cpp b/libosmscout-map/src/osmscoutmap/oss/Parser.cpp
index a5970450b..9736df775 100644
--- a/libosmscout-map/src/osmscoutmap/oss/Parser.cpp
+++ b/libosmscout-map/src/osmscoutmap/oss/Parser.cpp
@@ -2715,7 +2715,7 @@ void Errors::Warning(int line, int col, const char *s)
   error.column=col;
   error.text=s;
 
-  log.Warn() << error.line << "," << error.column << " " << "Warning: " << error.text;
+  //log.Warn() << error.line << "," << error.column << " " << "Warning: " << error.text;
 
   errors.push_back(error);
 }
@@ -2729,7 +2729,7 @@ void Errors::Warning(const char *s)
   error.column=0;
   error.text=s;
 
-  log.Warn() << error.line << "," << error.column << " " << "Warning: " << error.text;
+  //log.Warn() << error.line << "," << error.column << " " << "Warning: " << error.text;
 
   errors.push_back(error);
 }
