diff --git a/src/arvgvdevice.c b/src/arvgvdevice.c
index a29bacb..d569f6f 100644
--- a/src/arvgvdevice.c
+++ b/src/arvgvdevice.c
@@ -1366,6 +1366,7 @@ arv_gv_device_constructed (GObject *object)
 	io_data->socket = g_socket_new (G_SOCKET_FAMILY_IPV4,
 					G_SOCKET_TYPE_DATAGRAM,
 					G_SOCKET_PROTOCOL_UDP, NULL);
+	priv->io_data = io_data;
 	if (!g_socket_bind (io_data->socket, io_data->interface_address, FALSE, &local_error)) {
 		if (local_error == NULL)
 			local_error = g_error_new (ARV_DEVICE_ERROR, ARV_DEVICE_ERROR_UNKNOWN,
