diff --git a/libosmscout-client-qt/src/osmscout/DBJob.cpp b/libosmscout-client-qt/src/osmscout/DBJob.cpp
index e2b5a2d48..536abf217 100644
--- a/libosmscout-client-qt/src/osmscout/DBJob.cpp
+++ b/libosmscout-client-qt/src/osmscout/DBJob.cpp
@@ -107,7 +107,7 @@ void DBLoadJob::Run(const osmscout::BasemapDatabaseRef& basemapDatabase,
     }
     osmscout::GeoBox dbBox=db->GetDBGeoBox();
     if (!dbBox.Intersects(lookupBox)){
-      qDebug() << "Skip database" << db->path;
+      //qDebug() << "Skip database" << db->path;
       continue;
     }
     relevantDatabases.push_back(db);
